From da41258a7e5af338bc3a86d785643e7da2a35a01 Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <chrisg@marvell.com>
Date: Wed, 23 May 2018 12:10:36 +0300
Subject: [PATCH 0776/1200] fix: phy: marvell: fix handling of unconnected
 comphy

- the default value of comphy pipe selector is set to PCIe (x4)
  in case of unconnected comphy the default value remains 0x4
  which may lead to several issues with comphy initialization.

- this patch adds SMC call that powers off the comphy lane in case of
  unconnected comphy.

Change-Id: I367da8a7f439afaba5819c35fb0b1d4b843026af
Signed-off-by: Christine Gharzuzi <chrisg@marvell.com>
Signed-off-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/55504
Reviewed-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Igal Liberman <igall@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/57594
---
 drivers/phy/marvell/comphy_cp110.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/phy/marvell/comphy_cp110.c b/drivers/phy/marvell/comphy_cp110.c
index 07ff0c15de..8ed90ed9f4 100644
--- a/drivers/phy/marvell/comphy_cp110.c
+++ b/drivers/phy/marvell/comphy_cp110.c
@@ -548,6 +548,10 @@ int comphy_cp110_init(struct chip_serdes_phy_config *ptr_chip_cfg,
 		}
 		switch (ptr_comphy_map->type) {
 		case COMPHY_TYPE_UNCONNECTED:
+			ret = comphy_smc(MV_SIP_COMPHY_POWER_OFF,
+					 ptr_chip_cfg->comphy_base_addr,
+					 lane,
+					 ptr_comphy_map->type);
 		case COMPHY_TYPE_IGNORE:
 			continue;
 			break;
-- 
2.22.0

